chainer_chemistry/training/extensions/roc_auc_evaluator.py

```diff
@@ -0,0 +1,128 @@
+from chainer.dataset import convert
+from chainer.training.extensions import Evaluator
+from sklearn import metrics
+
+
+def _get_1d_numpy_array(v):
+    """Convert array or Variable to 1d numpy array
+    
+    Args:
+        v (numpy.ndarray or cupy.ndarray or chainer.Variable): array to be
+            converted to 1d numpy array 
+
+    Returns (numpy.ndarray): Raveled 1d numpy array
+
+    """
+    if isinstance(v, chainer.Variable):
+        v = v.data
+    return cuda.to_cpu(v).ravel()
+
+
+def _to_list(a):
+    """convert value `a` to list
+    
+    Args:
+        a: value to be convert to `list`
+
+    Returns (list):
+
+    """
+    if isinstance(a, (int, float)):
+        return [a, ]
+    else:
+        # expected to be list or some iterable class
+        return a
+
+
+class ROCAUCEvaluator(Evaluator):
+
+    """Evaluator which calculates ROC AUC score
+
+    Args:
+        iterator: Dataset iterator for the dataset to calculate ROC AUC score.
+            It can also be a dictionary of iterators. If this is just an 
+            iterator, the iterator is registered by the name ``'main'``.
+        target: Link object or a dictionary of links to evaluate. If this is
+            just a link object, the link is registered by the name ``'main'``.
+        converter: Converter function to build input arrays.
+            :func:`~chainer.dataset.concat_examples` is used by default.
+        device: Device to which the training data is sent. Negative value
+            indicates the host memory (CPU).
+        eval_hook: Function to prepare for each evaluation process. It is
+            called at the beginning of the evaluation. The evaluator extension
+            object is passed at each call.
+        eval_func: Evaluation function called at each iteration. The target
+            link to evaluate as a callable is used by default.
+        name (str): name of this extension. When `name` is None, 
+            `default_name='validation'` which is defined in super class 
+            `Evaluator` is used as extension name. This name affects to the
+            reported key name.
+        pos_labels (int or list): labels of the positive class, other classes
+            are considered as negative.
+        ignore_labels (int or list or None): labels to be ignored.
+            `None` is used to not ignore all labels.
+
+    Attributes:
+        converter: Converter function.
+        device: Device to which the training data is sent.
+        eval_hook: Function to prepare for each evaluation process.
+        eval_func: Evaluation function called at each iteration.
+        pos_labels (list): labels of the positive class
+        ignore_labels (list): labels to be ignored.
+
+    """
+
+    def __init__(self, iterator, target, predictor=None,
+                 converter=convert.concat_examples,
+                 device=None, eval_hook=None, eval_func=None, name=None,
+                 pos_labels=1, ignore_labels=-1):
```

Comment:
How do you feel adding `neg_label` argument?

Reply:
I wanted similar API with sklearn, which uses only `pos_label` (only one int is avaialble), and rest is considered as negative.